common/djangoapps/student/views/dashboard.py

```diff
@@ -531,6 +535,29 @@ def _credit_statuses(user, course_enrollments):
     return statuses
 
 
+def show_load_all_courses_link(user, load_all_courses, course_enrollments):
+    """
+    By default dashboard will show limited courses based on the course limit
+    set in configuration.
+
+    A link would be provided provided at the bottom to load all the courses if there are any courses.
+    """
+
+    total_enrollments = CourseEnrollment.enrollments_for_user(user).count()
+    if load_all_courses:
+        return False
```

Comment:
Nit (optional): You could move this check to the top so that we don't even do the `count()` if the `load_all_courses` is `True`.